x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/DistilBertTokenizer.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.ml.inference.nlp.tokenizers;
+
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.NlpConfig;
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.TokenizationParams;
+import org.elasticsearch.xpack.ml.inference.nlp.DistilBertRequestBuilder;
+import org.elasticsearch.xpack.ml.inference.nlp.NlpTask;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedMap;
+
+/**
+ * Specialization of the BERT Tokenizer for distilbert inference
+ */
+public class DistilBertTokenizer extends BertTokenizer {
+
+    private DistilBertTokenizer(List<String> originalVocab,
+                                SortedMap<String, Integer> vocab,
+                                boolean doLowerCase,
+                                boolean doTokenizeCjKChars,
+                                boolean doStripAccents,
+                                boolean withSpecialTokens,
+                                int maxSequenceLength,
+                                Set<String> neverSplit) {
+        super(originalVocab, vocab, doLowerCase, doTokenizeCjKChars, doStripAccents, withSpecialTokens, maxSequenceLength, neverSplit);
+    }
+
+    @Override
+    public NlpTask.RequestBuilder requestBuilder(NlpConfig config, NlpTask.ResultProcessorFactory factory) {
+        return new DistilBertRequestBuilder(this, factory);
```

Comment:
Is the only difference between `BertTokenizer` and `DistilBertTokenizer` in this method in that it returns a different request builder? If so can't be request builder be injected into the base `BertTokenizer` class possibly using another factory method. 


In NlpTokenizer build could become

```
    static NlpTokenizer build(Vocabulary vocabulary, TokenizationParams params) {
        ExceptionsHelper.requireNonNull(params, TOKENIZATION_PARAMS);
        ExceptionsHelper.requireNonNull(vocabulary, VOCABULARY);
        if (params instanceof BertTokenizationParams) {
            return BertTokenizer.builder(vocabulary.get(), params).requestBuilder(new BertRequestBuilder).build();
        }
        if (params instanceof DistilBertTokenizationParams) {
            return BertTokenizer.builder(vocabulary.get(), params).requestBuilder(new DistillBertRequestBuilder)build();
        }
        throw new IllegalArgumentException("unknown tokenization_params type [" + params.getName() + "]");
    }
```

It would be good if the tokeniser only knows about tokenising and needn't be the goto place for the request builder or the result processor.

Reply:
^^ Good idea. Will do